3scaleAdapter/pkg/threescale/threescale.go

```diff
@@ -86,27 +193,36 @@ func (s *Threescale) HandleAuthorization(ctx context.Context, r *authorization.H
 		return &result, err
 	}
 
-	status, err := s.isAuthorized(cfg, *r.Instance, systemC)
+	returnedStatus, err := s.isAuthorized(cfg, *r.Instance, systemC)
 	if err != nil {
 		log.Errorf("error authorizing request - %s", err.Error())
 	}
 
-	result.Status = status
+	result.Status = returnedStatus
+
+	// Add caching to valid calls.
+	if returnedStatus.Code == 0 {
+		seconds, _ := strconv.Atoi(cfg.CacheValidSeconds)
+		hits, _ := strconv.Atoi(cfg.CacheValidHits)
+		result.ValidDuration = time.Duration(seconds) * time.Second
+		result.ValidUseCount = int32(hits)
+	} else {
+		// No Cache for failed auth
+		result.ValidDuration = 0 * time.Second
+		result.ValidUseCount = 0
```

Comment:
Are both these lines required?

Reply:
Just forcing it right now, could be left by default and see how it works, still need to test it.